src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3574,78 +3574,170 @@ internal Binder GetBinder(CSharpSyntaxNode syntaxNode)
+                    var (currentGet, prevGet) = ((SourcePropertyAccessorSymbol?)currentProperty.GetMethod, (SourcePropertyAccessorSymbol?)prevProperty.GetMethod);
+                    if (currentGet != null || prevGet != null)
                     {
-                        diagnostics.Add(ErrorCode.ERR_PartialMethodMustHaveLatent, method.GetFirstLocation(), method);
+                        var accessorName = (currentGet ?? prevGet)!.Name.AsMemory();
+                        mergeAccessors(ref membersByName, accessorName, currentGet, prevGet);
                     }
-                    else if (method is { IsPartialDefinition: true, OtherPartOfPartial: null, HasExplicitAccessModifier: true })
+
+                    var (currentSet, prevSet) = ((SourcePropertyAccessorSymbol?)currentProperty.SetMethod, (SourcePropertyAccessorSymbol?)prevProperty.SetMethod);
+                    if (currentSet != null || prevSet != null)
                     {
-                        diagnostics.Add(ErrorCode.ERR_PartialMethodWithAccessibilityModsMustHaveImplementation, method.GetFirstLocation(), method);
+                        var accessorName = (currentSet ?? prevSet)!.Name.AsMemory();
+                        mergeAccessors(ref membersByName, accessorName, currentSet, prevSet);
                     }
+
+                    if ((object)membersByName == _lazyEarlyAttributeDecodingMembersDictionary)
+                    {
+                        // Avoid mutating the cached dictionary and especially avoid doing this possibly on multiple threads in parallel.
+                        membersByName = new Dictionary<ReadOnlyMemory<char>, ImmutableArray<Symbol>>(membersByName, ReadOnlyMemoryOfCharComparer.Instance);
+                    }
+
+                    membersByName[name] = FixPartialMember(membersByName[name], prevProperty, currentProperty);
                 }
-            }
 
-            memberNames.Free();
+                void mergeAccessors(ref Dictionary<ReadOnlyMemory<char>, ImmutableArray<Symbol>> membersByName, ReadOnlyMemory<char> name, SourcePropertyAccessorSymbol? currentAccessor, SourcePropertyAccessorSymbol? prevAccessor)
+                {
+                    Debug.Assert(currentAccessor != null || prevAccessor != null);
+
+                    // When an accessor is present on definition but not on implementation, the accessor is said to be missing on the implementation.
+                    // When an accessor is present on implementation but not on definition, the accessor is said to be unexpected on the implementation.
+                    if (currentAccessor == null)
+                    {
+                        // Partial definition is the source of truth for which accessors should be present.
+                        var (errorCode, propertyToBlame) = prevProperty.IsPartialDefinition ? (ErrorCode.ERR_PartialPropertyMissingAccessor, currentProperty) : (ErrorCode.ERR_PartialPropertyUnexpectedAccessor, prevProperty);
```

Comment:
Nit: maybe yet another local function? Fine if you prefer to not do this, but I just see more duplicate code except for which parameter order we use.

Reply:
I'll make an effort to condense this.

Reply:
FWIW, I considered this, but thought the amount of duplication/complexity was too minimal to justify a local function

Reply:
I have made an attempt to condense without introducing a local function